### PR TITLE
Remove prefixes from the IDs in the module templates

### PR DESCRIPTION
**Pull request by mjahoda**
https://github.com/redhat-documentation/modular-docs/issues/198

### CODE REVIEW

**Review by thatdocslady — COMMENTED** [review]
LGTM, do we also need to make any changes in the Reference Guide to reflect this guidance? We might have some examples that will need to be updated

**Comment by emmurphy1** [comment]
@mjahoda please update the comments in the template files and check the reference guide as well. For example, in the templates:
Base the file name and the ID on the module title. For example:
* file name: proc-doing-procedure-a.adoc
* ID: [id="proc-doing-procedure-a_{context}"]
* Title: = Doing procedure A

I

**Comment by mjahoda** [comment]
@emmurphy1 @thatdocslady Thank you for your hints. I attempted to discover all affected parts but there is still a chance that I have overlooked something. I also noticed two broken xrefs (and two incorrect examples on the same lines), so I fixed them within this PR as well. Please review once again.

**Review by thatdocslady — COMMENTED** [review]
I started commenting on each potential point but then I realized that I might be missing something in general, since the edits didn't seem to remove the module/assembly prefix at all, and in fact some of them added prefixes where there were none before. Maybe I'm mistaken, but it's worth another look though the changes..

**Comment by mjahoda** [comment]
> the edits didn't seem to remove the module/assembly prefix at all, and in fact some of them added prefixes where there were none before

@thatdocslady I double-checked the diff [1] and also the individual commits, and I see only removals of prefixes from IDs. Could you please provide an example (a link or a screenshot from the web UI) of the described problem? Because I'm so used to do everything in Gitlab, there is still a chance that I'm messing something up here on Github.

[1] https://github.com/redhat-documentation/modular-docs/pull/201/files

File: modular-docs-manual/content/topics/module_creating-assemblies.adoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ An assembly could represent an article, "chapter" in a book, or even an entire b
 . Create an assembly file for each user story that you identified.
 +
 --
-Be sure to follow the conventions for xref:anchor-and-file-names[naming anchors and files]. For example: `guided-decision-tables.adoc`
+Follow the conventions for xref:module_anchor-and-file-names-concept[naming anchors and files]. For example: `assembly_designing-guided-decision-tables.adoc`
```
Comment (thatdocslady) [inline modular-docs-manual/content/topics/module_creating-assemblies.adoc:13]:
```suggestion
Follow the conventions for xref:anchor-and-file-names-concept[naming anchors and files]. For example: `designing-guided-decision-tables.adoc`
```

Comment (thatdocslady) [reply modular-docs-manual/content/topics/module_creating-assemblies.adoc:13]: This looks like it added a prefix where there was none. AFAIK we are implementing the prefix removal for everything including assemblies, but please correct me if I'm wrong!

File: modular-docs-manual/content/topics/module_creating-modules.adoc [hunk 2]
```diff
@@ -10,7 +10,7 @@ After identifying and creating the assemblies, each assembly should identify the
 --
 Each assembly should already have the names of the modules that should be included in the assembly. Now you just need to create the actual files for those modules.
 
-Be sure to follow the conventions for xref:anchor-and-file-names[naming anchors and files]. For example: `guided-decision-tables.adoc`
+Follow the conventions for xref:module_anchor-and-file-names-concept[naming anchors and files]. For example: `proc_creating-guided-decision-tables.adoc`
```
Comment (thatdocslady) [inline modular-docs-manual/content/topics/module_creating-modules.adoc:13]:
```suggestion
Follow the conventions for xref:anchor-and-file-names-concept[naming anchors and files]. For example: `creating-guided-decision-tables.adoc`
```

File: modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ////
 Base the file name and the ID on the module title. For example:
-* file name: con-my-concept-module-a.adoc
-* ID: [id="con-my-concept-module-a_{context}"]
+* file name: con_my-concept-module-a.adoc
```
Comment (thatdocslady) [inline modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc:3]:
```suggestion
* file name: my-concept-module-a.adoc
```

File: modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ////
 Base the file name and the ID on the module title. For example:
-* file name: proc-doing-procedure-a.adoc
-* ID: [id="proc-doing-procedure-a_{context}"]
+* file name: proc_doing-procedure-a.adoc
```
Comment (thatdocslady) [inline modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc:3]:
```suggestion
* file name: doing-procedure-a.adoc
```

**Review by thatdocslady — COMMENTED** [review]
Sorry @mjahoda if this wasn't clear, here is a screenshot of what I'm seeing in the diff viewer:
Screenshot 2023-04-24 at 12 59 17 1

You can see a side-by-side view of the various files, and the green boxes are meant to represent the proposed changes. I went ahead and proposed suggestions inline in the places where I noticed the issues, for example a prefix was actually added rather than removed, or simply changed from dash to underscore, etc.

Hopefully I'm not misreading the whole thing, but IIRC the decision was to remove prefixes from IDs and file names for all module types and assemblies. Please correct me if I'm wrong or missing anything, GH and GL have similar web UIs but some things are indeed different so they might not be skimmable in the same way (no worries, it's just practice!)

**Comment by mjahoda** [comment]
@thatdocslady Now, I see what probably confused you. Within this PR, I also fixed some other things such as broken xrefs. However, I did it in the most straightforward way (fixed an ID according to the existing xrefs, not the other way around). In the internal structure of the mod-docs ref. guide, we have really many things that no longer reflects what we recommend in the guide itself. Fixing all of it will be time-consuming and I believe it's not necessary. The important parts are the published content, the examples, and the template files.

Then, we are still **preserving the prefixes in the file names**.

Given that, I believe that the goal of this PR - remove prefixes from IDs - is met.

File: modular-docs-manual/content/topics/module_creating-assemblies.adoc (hunk 1 above)
Comment (mjahoda) [reply modular-docs-manual/content/topics/module_creating-assemblies.adoc:13]: Yes, this is just a (bit unrelated) fix of an xref (in a more convenient way). And regarding the example - we preserve prefixes in file names.

File: modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc (hunk 3 above)
Comment (mjahoda) [reply modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc:3]: We preserve prefixes in file names.

File: modular-docs-manual/files/TEMPLATE_REFERENCE_reference-material.adoc [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ////
 Base the file name and the ID on the module title. For example:
-* file name: ref-my-reference-a.adoc
-* ID: [id="ref-my-reference-a_{context}"]
+* file name: ref_my-reference-a.adoc
```
[the remark this answers is not in the document]
Comment (mjahoda) [reply modular-docs-manual/files/TEMPLATE_REFERENCE_reference-material.adoc:3]: We preserve prefixes in file names.

File: modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc (hunk 4 above)
Comment (mjahoda) [reply modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc:3]: We preserve prefixes in file names.

**Review by ncbaratta — APPROVED** [review]
lgtm

File: modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (mjahoda) [reply modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc:3]: @emmurphy1 I think that it makes sense to distinguish the prefix part and the title part of a file name this way. Moreover, it's a default in the newdoc tool, all RHEL modules and assemblies follow this pattern, and if you check the source files [1] of our mod-docs guide... :-)

[1] https://github.com/redhat-documentation/modular-docs/tree/main/modular-docs-manual/content/topics

**Pull request merged**